be/src/util/mem_info.cpp

```diff
@@ -34,39 +34,70 @@
 
 #include "util/mem_info.h"
 
+
 #include <cstdlib>
 #include <fstream>
 #include <iostream>
+#include <linux/magic.h>
+#include <sys/vfs.h>
 
 #include "fs/fs_util.h"
 #include "gutil/strings/split.h"
+#include "util/errno.h"
 #include "util/pretty_printer.h"
 #include "util/string_parser.hpp"
 
 namespace starrocks {
 
+#ifndef CGROUP2_SUPER_MAGIC
+#define CGROUP2_SUPER_MAGIC 0x63677270
+#endif
+
 bool MemInfo::_s_initialized = false;
 int64_t MemInfo::_s_physical_mem = -1;
 
 void MemInfo::init() {
     // check if application is in docker container or not via /.dockerenv file
-    if (fs::path_exist("/.dockerenv")) {
-        // Read from /sys/fs/cgroup/memory/memory.limit_in_bytes
-        std::ifstream memoryLimit("/sys/fs/cgroup/memory/memory.limit_in_bytes");
-        std::string line;
+    while (fs::path_exist("/.dockerenv")) {
```

Comment:
Seems this will work even in bare metals. So if we can get memory information from it, we don't need to read `/proc/meminfo`.
And you can extract this into a function, then you don't need `while`

Reply:
in bare metals, the /sys/fs/cgroup/memory/memory.limit_in_bytes may be int64_t::max(). So still need to get /proc/meminfo.